src/Disciplina/WorldState/Mode.hs

```diff
@@ -0,0 +1,59 @@
+{-# LANGUAGE TemplateHaskell #-}
+
+-- | Module contains the definition of Witness's WorkMode and its implementations.
+
+module Disciplina.WorldState.Mode
```

Comment:
Probably it's better to have `Disciplina.Witness.Launcher.Mode` for sake of uniformity. (Also, yes, you can additionally rename the whole `Disciplina.WorldState` to `Disciplina.Witness`, it would be nice).

Reply:
:tada: